build/tasks/GenerateRestoreSourcesPropsFile.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.IO;
+using System.Xml;
+using System.Xml.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System.Text;
+
+namespace RepoTasks
+{
+    public class GenerateRestoreSourcesPropsFile : Task
+    {
+        [Required]
+        public ITaskItem[] Sources { get; set; }
+
+        [Required]
+        public string OutputPath { get; set; }
+
+        public override bool Execute()
+        {
+            OutputPath = OutputPath.Replace('\\', '/');
+            Directory.CreateDirectory(Path.GetDirectoryName(OutputPath));
+
+            var sources = new XElement("DotNetRestoreSources");
+            var doc = new XDocument(new XElement("Project", new XElement("PropertyGroup", sources)));
+
+            var sb = new StringBuilder();
+
+            sb.AppendLine();
+            foreach (var source in Sources)
+            {
+                sb.Append("      ").Append(source.ItemSpec).AppendLine(";");
```

Comment:
Certainny an odd way to do this. Let's at least use a variable to store the indent.

Reply:
Updated.